test/plugin.js

```diff
@@ -61,7 +61,7 @@ describe('Plugin tests', function() {
     var plugin;
 
     beforeEach(function() {
-      plugin = new Plugin(new Session(new Connection, 'id'), 'name', 'id');
+      plugin = new Plugin(new Session(new Connection('id', {address: 'address'}), 'id'), 'name', 'id');
```

Comment:
Well mocking methods is always the same work to do, plus you don't need to adjust anything unless the mocked method changes. This is crucial to not over-adjust when changing tests. Given you *mock* similar way in multiple tests, constructor changes will bring loads of unrelated specs changes.

Just something to consider as it's just different (imo worse only in very short run) point of view.